### PR TITLE
docs: add vxe-table doc

### PR DESCRIPTION
**Pull request by ArthurDarkstone**
## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [x] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

## Checklist

> ℹ️ Check all checkboxes - this will indicate that you have done everything in accordance with the rules in [CONTRIBUTING](contributing.md).

- [x] If you introduce new functionality, document it. You can run documentation with `pnpm run docs:dev` command.
- [x] Run the tests with `pnpm test`.
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

## Release Notes

- **New Features**
  - Introduced dark mode toggle with improved theme management.
  - Added new Vue components for enhanced table functionalities, including editable cells, custom cell rendering, tree tables, virtual scrolling, and remote data fetching.
  - Implemented a mock API for simulating data fetching.
  - Enhanced grid options with features like pagination, sorting, and dynamic data loading.
  - Improved caching strategy for service worker configurations.

- **Documentation**
  - Expanded and clarified documentation for the Vben Vxe Table component, including detailed examples and configurations.

- **Bug Fixes**
  - Improved handling of AJAX requests and pagination in various components.

- **Chores**
  - Updated dependencies to enhance performance and compatibility.

### CODE REVIEW

**Comment by vince292007** [comment]
ci有报错

**Comment by ArthurDarkstone** [comment]
> ci有报错

是的，本地构建发现是`@vben/icons` 中引入了 `DOMParser`，而 vitepress 是node服务端渲染导致报错。`ReferenceError: DOMParser is not defined`

File: docs/.vitepress/theme/index.ts [hunk 1]
```diff
@@ -15,11 +20,71 @@ import 'virtual:group-icons.css';
...
 
 export default {
-  enhanceApp(ctx: EnhanceAppContext) {
+  async enhanceApp(ctx: EnhanceAppContext) {
     const { app } = ctx;
     app.component('VbenContributors', VbenContributors);
     app.component('DemoPreview', DemoPreview);
     app.use(NolebaseGitChangelogPlugin);
+
+    if (!import.meta.env.SSR) {
```
Comment (ArthurDarkstone) [inline docs/.vitepress/theme/index.ts:29]: 由于在vitepress 中使用了vxe-table, 导致构建报错 `window is not defined` 类似错误，用此种方法引入可以避免
详见 [vitepress SSR 兼容](https://vitepress.dev/zh/guide/ssr-compat#conditional-import)

Comment (ArthurDarkstone) [reply docs/.vitepress/theme/index.ts:29]: @vince292007 用这种方式先处理了下，如果有更好的方法欢迎补充

**Pull request merged**